src/abstractdataframe/reshape.jl

```diff
@@ -65,49 +70,75 @@ d1 = DataFrame(a = repeat([1:3;], inner = [4]),
+               id_vars::Vector{Int}; variable_name::Symbol=:variable,
+               value_name::Symbol=:value)
     N = length(measure_vars)
     cnames = names(df)[id_vars]
-    insert!(cnames, 1, :value)
-    insert!(cnames, 1, :variable)
+    insert!(cnames, 1, value_name)
+    insert!(cnames, 1, variable_name)
     DataFrame(Any[Compat.repeat(_names(df)[measure_vars], inner=nrow(df)),   # variable
                   vcat([df[c] for c in measure_vars]...),                    # value
                   [Compat.repeat(df[c], outer=N) for c in id_vars]...],      # id_var columns
               cnames)
 end
-function stack(df::AbstractDataFrame, measure_var::Int, id_var::Int)
-    stack(df, [measure_var], [id_var])
+function stack(df::AbstractDataFrame, measure_var::Int, id_var::Int;
+               variable_name::Symbol=:variable, value_name::Symbol=:value)
+    stack(df, [measure_var], [id_var];
+          variable_name=variable_name, value_name=value_name)
+end
+function stack(df::AbstractDataFrame, measure_vars::Vector{Int}, id_var::Int;
+               variable_name::Symbol=:variable, value_name::Symbol=:value)
+    stack(df, measure_vars, [id_var];
+          variable_name=variable_name, value_name=value_name)
 end
-function stack(df::AbstractDataFrame, measure_vars::Vector{Int}, id_var::Int)
-    stack(df, measure_vars, [id_var])
+function stack(df::AbstractDataFrame, measure_var::Int, id_vars::Vector{Int};
+               variable_name::Symbol=:variable, value_name::Symbol=:value)
+    stackdf(df, [measure_var], id_vars;
+            variable_name=variable_name, value_name=value_name)
 end
-function stack(df::AbstractDataFrame, measure_var::Int, id_vars::Vector{Int})
-    stackdf(df, [measure_var], id_vars)
+function stack(df::AbstractDataFrame, measure_vars, id_vars;
+               variable_name::Symbol=:variable, value_name::Symbol=:value)
+    stack(df, index(df)[measure_vars], index(df)[id_vars];
+          variable_name=variable_name, value_name=value_name)
 end
-stack(df::AbstractDataFrame, measure_vars, id_vars) =
-    stack(df, index(df)[measure_vars], index(df)[id_vars])
 # no vars specified, by default select only numeric columns
-numeric_vars(df::AbstractDataFrame) = [T <: AbstractFloat || (T <: Nullable && eltype(T) <: AbstractFloat)
-                                       for T in eltypes(df)]
-function stack(df::AbstractDataFrame, measure_vars = numeric_vars(df))
+numeric_vars(df::AbstractDataFrame) =
+    [T <: AbstractFloat || (T <: Nullable && eltype(T) <: AbstractFloat)
+     for T in eltypes(df)]
+
+function stack(df::AbstractDataFrame, measure_vars = numeric_vars(df);
+               variable_name::Symbol=:variable, value_name::Symbol=:value)
     mv_inds = index(df)[measure_vars]
-    stack(df, mv_inds, _setdiff(1:ncol(df), mv_inds))
+    stack(df, mv_inds, _setdiff(1:ncol(df), mv_inds);
+          variable_name=variable_name, value_name=value_name)
 end
 
 """
```

Comment:
If you feel like it, it would be nice to remove this docstring in favor of the one from above, splitting `stack` and `melt`. Another option would be to use exactly the same docstring for both, I think some doc tricks allow doing this.